### DISCUSSION TITLE
No matching distribution found for AutoRA core

### DISCUSSION THREAD

**Discussion by musslick** [AutoResearch/autora#364] [Core Contributions] [answered]
After cloning a repository that was created from the `autora-theorist-template`, I followed the instructions in the Readme to set up the environment.
After running `pip install -e ".[dev]"`, I  get the following error:

```
ERROR: Ignored the following versions that require a different python version: 3.0.0a0 Requires-Python >=3.8.10,<3.11; 3.0.0a1 Requires-Python >=3.8.10,<3.11; 3.0.0a2 Requires-Python >=3.8.10,<3.11; 3.0.0a3 Requires-Python <3.11,>=3.8.10
ERROR: Could not find a version that satisfies the requirement autora-core (from autora-theorist-template[dev]) (from versions: none)
ERROR: No matching distribution found for autora-core

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip
(venv) (base) PNI-20C02DQ02WML88:autora-experimentalist-template musslick$ 
```
@hollandjg @younesStrittmatter Any ideas on how to resolve this?

This is possibly related to #363

**Comment by younesStrittmatter** [comment]
The error suggests that it has to do with your python version. What python version are you running in your environment? Maybe you have to create your environment with a specific version

**Comment by hollandjg** [comment]
I thought the same. Can you try
```shell
virtualenv venv -p python3.8
```
... before running the install command?

**Reply by musslick** [reply]
Unfortunately, that did not solve it. I still get the following error message:

```
(venv) (base) PNI-20C02DQ02WML88:autora-experimentalist-sampler-template musslick$ pip install -e ".[dev]"
Obtaining file:///Users/musslick/Documents/recovered%20files/projects/AdversarialResearch/AutoRA%20packages/autora-experimentalist-sampler-template
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Installing backend dependencies ... done
  Preparing editable metadata (pyproject.toml) ... done
ERROR: Ignored the following versions that require a different python version: 3.0.0a0 Requires-Python >=3.8.10,<3.11; 3.0.0a1 Requires-Python >=3.8.10,<3.11; 3.0.0a2 Requires-Python >=3.8.10,<3.11; 3.0.0a3 Requires-Python <3.11,>=3.8.10
ERROR: Could not find a version that satisfies the requirement autora-core (from autora-theorist-template[dev]) (from versions: none)
ERROR: No matching distribution found for autora-core

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip

```

**Reply by younesStrittmatter** [reply]
Still seems like a problem with the python version. When running python in your command line, what version shows up?

**Reply by musslick** [reply, marked as answer]
You are right, it was the python version. Apparently installing `virtualenv venv -p python3.8` gave me Python version 3.8.2 which is not sufficient. What solved it was `virtualenv venv -p python3.9`. I added a pull request for changing the documentation to include the python version as you suggested below!

**Comment by younesStrittmatter** [comment]
we should definitely add this to the README. I think by default virtualenv uses the newest version.
